src/khoj/database/models/__init__.py

```diff
@@ -69,6 +72,35 @@ class Type(models.TextChoices):
     renewal_date = models.DateTimeField(null=True, default=None, blank=True)
 
 
+class ChatModelOptions(BaseModel):
+    class ModelType(models.TextChoices):
+        OPENAI = "openai"
+        OFFLINE = "offline"
+
+    max_prompt_size = models.IntegerField(default=None, null=True, blank=True)
+    tokenizer = models.CharField(max_length=200, default=None, null=True, blank=True)
+    chat_model = models.CharField(max_length=200, default="mistral-7b-instruct-v0.1.Q4_0.gguf")
+    model_type = models.CharField(max_length=200, choices=ModelType.choices, default=ModelType.OFFLINE)
+
+
+class Agent(BaseModel):
+    creator = models.ForeignKey(KhojUser, on_delete=models.CASCADE, default=None, null=True, blank=True)
+    name = models.CharField(max_length=200)
+    tuning = models.TextField()
+    avatar = models.URLField(max_length=400, default=None, null=True, blank=True)
+    tools = models.JSONField(default=list)  # List of tools the agent has access to, like online search or notes search
+    public = models.BooleanField(default=False)
+    managed_by_admin = models.BooleanField(default=False)
+    chat_model = models.ForeignKey(ChatModelOptions, on_delete=models.CASCADE)
+
+
+@receiver(pre_save, sender=Agent)
+def check_public_name(sender, instance, **kwargs):
+    if instance.public:
+        if Agent.objects.filter(name=instance.name, public=True).exists():
```

Comment:
Should we just enforce unique Agent names per creator. Agent public identifiers (like url, name) can be  [creator_name]/[agent_name] similar to how github, huggingface etc do naming conventions for repositories, models. For e.g an agent called `smith` created by username = `neo` will be called `neo/smith` etc.

```suggestion
        if Agent.objects.filter(name=instance.name, creator=instance.creator).exists():
```

Reply:
Hmm, the tricky bit about that is that the `given_name`/`family_name` fields on the user objects are nullable, and email address wouldn't make for a good route. I get your point, but I think let's optimize this later on. It's not a concern until we expose the ability to make new agents.